### ISSUE TITLE
docker stack 3.2 support endpoint_mode: dnsrr not applied

### ISSUE THREAD

**Issue by webfrank** [moby/moby#33276]
**Description**

On docker 17.04.00 creating a compose file with endpoint_mode: dnsrr and deploying with docker stack deploy the desired mode is not applied, always vip.

**Steps to reproduce the issue:**
1. yawl file with service definition like this:
view-web:
    image: docker.registry.xxx/view-web
    deploy:
      mode: replicated
      replicas: 1
      endpoint_mode: dnsrr
      placement:
        constraints:
          - node.role == manager
    networks:
      - view-network
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
2. docker stack deploy

**Describe the results you received:**

docker service inspect reports:

            "Networks": [
                {
                    "Target": "xcg1u4dycdtlke7s3h9spmyku",
                    "Aliases": [
                        "view-web"
                    ]
                }
            ],
            "EndpointSpec": {
                "Mode": "vip"
            }
        },
        "Endpoint": {
            "Spec": {
                "Mode": "vip"
            },

**Describe the results you expected:**

        "Endpoint": {
            "Spec": {
                "Mode": "dnsrr"
            }
        },

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client:
 Version:      17.05.0-ce
 API version:  1.29
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 22:10:54 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.05.0-ce
 API version:  1.29 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 22:10:54 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Output of `docker info`:**

```
Containers: 17
 Running: 5
 Paused: 0
 Stopped: 12
Images: 138
Server Version: 17.05.0-ce
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 611
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: active
 NodeID: y3h1aeht7xf7zdhn6obueh6up
 Is Manager: true
 ClusterID: lbx9l3qb0nllni7uknocpg1sz
 Managers: 1
 Nodes: 1
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
 Node Address: 192.168.0.166
 Manager Addresses:
  192.168.0.166:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 9048e5e50717ea4497b757314bad98ea3763c145
runc version: 9c2d8d184e5da67c95d601382adf14862e4f2228
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-63-generic
Operating System: Ubuntu 16.04.2 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 3.859GiB
Name: view2-master
ID: VVFF:ILND:QOXX:BPXX:MYG5:63CE:MCNL:IDFE:EV7E:WG46:WJ2J:SHGD
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by webfrank** [comment]
updating to 17.05.00 results in:

 "EndpointSpec": {
                "Mode": "dnsrr"
            }

seems ok

**Comment by Fabryprog** [comment]
This feature is available from "3.2" compose version 

Upgrade your docker compose version

**Comment by dnephin** [comment]
Sounds like this was fixed, so closing.
